### ISSUE TITLE
 Error: GraphQL request was redirected to /

### ISSUE THREAD

**Issue by aryanrajput** [AxeWP/wp-graphql-rank-math#15]
### Description

I'm able to get seo field on WordPress GraphQL IDE.  But on gatsby develop giving this error.

### Steps to reproduce

Getting this Error: GraphQL request was redirected to /

### Additional Context

![image](https://user-images.githubusercontent.com/26869121/199181806-1985a006-fa74-450c-a677-721f8b35cf10.png)

![image](https://user-images.githubusercontent.com/26869121/199181429-c69241a0-c22d-4f16-8282-32197e908587.png)

### Plugin Version

WPGraphQL for Rank Math SEO Version 0.0.5

### WordPress Version

6.0.3

### WPGraphQL Version

1.12.0

### RankMath SEO Version

1.0.100.1

### Additional enviornmental details

"react": "^18.1.0",
"gatsby": "^4.19.2",
"gatsby-source-wordpress": "^6.24.0",

{
      resolve: `gatsby-source-wordpress`,
      options: {
        schema: {
          timeout: 100000,
        },
        url: `https://cliniexpertsresearch.net/graphql`,
      }
    },

### Please confirm that you have searched existing issues in the repo.

- [X] Yes

### Please confirm that you have disabled ALL plugins except for RankMath SEO, WPGraphQL, and WPGraphQL for Rank Math

- [ ] Yes
- [ ] My issue is with a specific 3rd-party plugin.

**Comment by justlevine** [comment]
Hey @aryanrajput, I need this part of the bug report 😇:
> Please confirm that you have disabled ALL plugins except for RankMath SEO, WPGraphQL, and WPGraphQL for Rank Math

All those other plugins need to be disabled so we can confirm whether the issue is actually with this plugin and we have the power to fix/address it.

If the issue goes away, when disabling all other plugins/custom php, enable them _one by one_ until the error returns, and we've found the culprit. It could be an issue with that plugin/custom code specifically or a conflict between that one and WPGraphQL for RankMath, but either way we have a good starting point for replicating and resolving the issue.

**Comment by aryanrajput** [comment]
Same result.

![image](https://user-images.githubusercontent.com/26869121/199221562-9f332c24-528d-4cc4-a947-1026d0e4ae45.png)

![image](https://user-images.githubusercontent.com/26869121/199221659-8d56e25b-929d-4893-bfc7-30f39cb9478e.png)

![image](https://user-images.githubusercontent.com/26869121/199221737-7b8a5120-f565-4365-9645-b8a581e5d9e3.png)

**Comment by justlevine** [comment]
Thanks for the followup.

🤔

Running `gatsby develop` is working fine for me... When  and we're not actually handling image meta yet in `v0.0.5`, so I'm not sure what could be causing this.

If you disable WPGraphQL for Rank Math, does the problem go away?, Similarly does the issue persist if Gatsby is building from a clean WP install instead of your existing site?

**Comment by aryanrajput** [comment]
Yes problem will go away after disable WPGraphQL-for-Rank-Math plugin.

And I have tested same WP on new Gatsby project. as expected result is same.

![image](https://user-images.githubusercontent.com/26869121/199251321-d9f97578-48d8-40e2-a496-61767940bebb.png)

**Comment by aryanrajput** [comment]
@justlevine No Response get from your end. Please R&D and let me know the exact reason.

**Comment by justlevine** [comment]
@aryanrajput i've tried and tried to replicate using [Gatsby Starter WordPress Homepage](https://www.gatsbyjs.com/starters/gatsbyjs/gatsby-starter-wordpress-homepage), but Gatsby is building fine for me.

![image](https://user-images.githubusercontent.com/29322304/200651531-c3b1592c-8ba0-4b7e-8d4b-87d5e00b0633.png)
![image](https://user-images.githubusercontent.com/29322304/200651615-7efc7851-2c2a-4ce6-b77c-014f5423a495.png)

While investigating, I _did_ find some issues with `seo.fullHead` and `seo.openGraph` in certain environments, and pushed a [new release](https://github.com/AxeWP/wp-graphql-rank-math/releases/tag/0.0.6), but I'm doubtful it will help your issue.

Would you be able to create a test replication on CodeSandbox, and share here?

**Comment by aryanrajput** [comment]
Can you please provide your email so that I can share my graphQL endpoint. that you can use it in your project to check whether it is Gatsby end problem or WordPress end.

**Comment by justlevine** [comment]
support[at]axepress.dev

**Comment by aryanrajput** [comment]
sent.

**Comment by justlevine** [comment]
Hey @aryanrajput 
I was able to replicate the error on your endpoint. This is indeed an issue with your WordPress data. 

Your endpoint started timing out before I could look too deep but most likely:
- You have custom code or a plugin which changes how redirects are handled.
- You have a corrupt image in your Media Library.
- One of your posts has a Featured Image that has since been deleted from the Media Library.

A confluence of all three _might_ be part of Rank Math's Redirections module. If you disable the `Redirections` module in Rank Math, does gatsby build?

cc: https://github.com/gatsbyjs/gatsby/discussions/35331

**Comment by aryanrajput** [comment]
Thanks @justlevine . problem solved. 
Just disable this option and everything solved.

![image](https://user-images.githubusercontent.com/26869121/201617068-34d3cd30-c40e-4608-afde-96fa23a766d8.png)

**Comment by aryanrajput** [comment]
Hi @justlevine , 

Showing duplicate data in SEO fullHead field.

![image](https://user-images.githubusercontent.com/26869121/208365672-7ab0a13c-16e2-43ba-bbb2-f9ee9087ef21.png)

First item field showing normal but as index increase fullHead field data also increase accordingly.

**Comment by justlevine** [comment]
@aryanrajput can you confirm you're running the latest version? I thought I squashed that in v0.0.6 🤔

**Comment by aryanrajput** [comment]
![image](https://user-images.githubusercontent.com/26869121/208381115-e26beac3-f522-4121-abf4-28a466a75329.png)

**Comment by justlevine** [comment]
Please upgrade to the latest version 0.0.7 and confirm if the issue goes away

**Comment by aryanrajput** [comment]
Hi @justlevine currently getting this result.

![image](https://user-images.githubusercontent.com/26869121/208584457-4a4206e0-5dca-4f72-94d6-6798a17e5554.png)

I don't think the issue has been solved. you can see this IDE screenshot. 

![image](https://user-images.githubusercontent.com/26869121/208584722-c3523cac-b9fb-4cd7-a4a4-102fda90156a.png)

![image](https://user-images.githubusercontent.com/26869121/208584818-ef1c2b50-5933-473a-ac0e-a6565d9bf23c.png)

Sometimes on WP-GraphQL IDE . getting this error.
![image](https://user-images.githubusercontent.com/26869121/208585018-a0a452c0-c1d6-419a-aa93-64cdd3c394b2.png)

**Comment by aryanrajput** [comment]
Hi @justlevine ,

did you find out the issue?

**Comment by justlevine** [comment]
@aryanrajput #22 (live on /develop) should resolve the issue with duplicate/old meta tags. Feel free to test and confirm.

The errors in your second screenshot are caused by an old version of WPGraphQL for Woocommerce being incompatible with WPGraphQL v1.13.x , upgrading that plugin should solve that.

**Comment by aryanrajput** [comment]
@justlevine Thank you for your support. 

Could you please give us a short of idea When are you planning to release the new version with this bug fix, so that we can plan accordingly.

**Comment by justlevine** [comment]
Blocked by #30 and #31

**Comment by yeasinshakil** [comment]
**

> Just disable Rank Math general setting Redirects attachment. 404 error or redirects attachment error will gone.

**

![Screenshot (383)](https://github.com/AxeWP/wp-graphql-rank-math/assets/106692233/aa1c755c-52a7-4cc7-b681-a79e04ce10fc)

**Comment by justlevine** [comment]
## 🤖 Automated Triage

**Classification:** type: bug 🐛 · priority: medium · status: hold :warning:
**Agent-feasible:** No — blocked by #30 and #31, and the root cause involves WordPress redirect behavior interacting with headless GraphQL consumption. Needs architectural direction first.
**Estimated effort:** Agent N/A · Human high

**Summary:** The RankMath "Redirect Attachments" setting causes headless consumers (e.g. gatsby-source-wordpress) to hit redirects during schema introspection. A workaround (disabling the redirect setting) is documented in comments. Core fix is blocked on upstream restructuring (#30, #31).

**Next steps:** Stays on hold pending #30/#31. The workaround in comments should be surfaced into README/docs to reduce duplicate reports.
